packages/api/src/routes/medical/internal-facility.ts

```diff
@@ -0,0 +1,143 @@
+import { metriportCompanyDetails } from "@metriport/shared";
+import { Request, Response } from "express";
+import Router from "express-promise-router";
+import httpStatus from "http-status";
+import { z } from "zod";
+import { createFacility } from "../../command/medical/facility/create-facility";
+import { getOrganizationOrFail } from "../../command/medical/organization/get-organization";
+import {
+  createOrUpdateCQOrganization,
+  getCqOrganization,
+} from "../../external/carequality/command/cq-directory/create-or-update-cq-organization";
+import { metriportEmail as metriportEmailForCq } from "../../external/carequality/constants";
+import { requestLogger } from "../helpers/request-logger";
+import { required } from "../schemas/shared";
+import { getUUIDFrom } from "../schemas/uuid";
+import { asyncHandler } from "../util";
+import { AddressStrictSchema } from "./schemas/address";
+
+const router = Router();
+
+export const cqOboOrgDetailsSchema = z.object({});
+export type CqOboOrgDetails = z.infer<typeof cqOboOrgDetailsSchema>;
+
+const facilityOboDetailsSchemaBase = z
+  .object({
+    name: z.string(),
+    npi: z.string(),
+    lat: z.string(),
+    lon: z.string(),
+    // CQ
+    cqActive: z.boolean().optional(),
+    cqOboOid: z.string().optional(),
+    // TODO 1706: implement this
+    // CW
+    // cwActive: z.boolean().optional(),
+    // cwOboOid: z.string().optional(),
+  })
+  .merge(AddressStrictSchema);
+type FacilityOboDetails = z.infer<typeof facilityOboDetailsSchemaBase>;
+
+const facilityOboDetailsSchema = facilityOboDetailsSchemaBase.refine(
+  required<FacilityOboDetails>("cqOboOid").when("cqActive"),
+  {
+    message: "cqObOid is required and can't be empty when cqActive is true",
+    path: ["cqObOid"],
+  }
+);
+
+/** ---------------------------------------------------------------------------
+ *
+ * PUT /internal/facility/
+ *
+ * Creates a new facility and registers it with HIEs the facility is enabled at.
+ *
+ * WIP: This endpoint is a work in progress and is not yet fully implemented.
+ *
+ * @return {FacilityDTO} The updated facility.
+ */
+router.put(
+  "/",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const facilityInput = facilityOboDetailsSchema.parse(req.body);
+    // TODO 1706 search existing facility by NPI, cqOboOid, and cwOboOid (individually), and update if exists
+
+    let facilityName: string | undefined;
```

Comment:
I think this should always be definted, and we should throw an error if the org entry doesn't have a name